### DISCUSSION TITLE
[RFC] Import fallbacks for easier module mocking

### DISCUSSION THREAD

**Discussion by shilman** [storybookjs/storybook#27473] [RFC]
**Status: implemented [typesafe module mocking](https://storybook.js.org/blog/type-safe-module-mocking/) in 8.2, possibly superseded by https://github.com/storybookjs/storybook/discussions/30877**

### Summary

A more convenient way to do module mocking that does not require editing `package.json` every time you want to mock.

### Problem Statement

Storybook recently introduced [type-save module mocking](https://storybook.js.org/blog/type-safe-module-mocking/) using a Node standard called `package.json` import subpaths.

The feature is great because it allows you to mock out any module in your Storybook, and this will work in Node, the browser, [portable stories](https://storybook.js.org/blog/portable-stories-for-playwright-ct/) in other tools, and so on.

In its current form, this approach requires adding a new entry to your `package.json` for every file you want to mock. For example, to mock the module at `./lib/settings.ts` you would add:

```json
{
  "imports": {
    "#lib/settings": {
      "storybook": "./lib/settings.mock.ts",
      "default": "./lib/settings.ts"
    },
    // etc.
  }
}
```

This works fine, but it's extra work to have to add/remove settings from `package.json` every time you change what you're mocking. It also clutters the `package.json` making it harder for people to read.

### Implementation

Webpack and Typescript support a fallback mechanism that makes this much easier.

```json
{
  "imports": {
    "#*": {
      "storybook": ["./*.mock", "./*"],
      "default": "./*"
    }
  }
}
```

Using this construct, users could just add this single entry to their `package.json` and then use as many `.mock` files as they need around their projects.

Node doesn't allow fallbacks for imports for perf reasons. And Vite doesn't allow it to follow the Node implementation. However, we believe that for non-production use cases, the performance argument does not apply, and that both Node and Vite should support this fallback syntax, possibly enabled by a configuration flag. We should lobby both tools to extend the standard.

### Deliverables

N/A

**Comment by monday-yi** [comment]
@ythx-101 收到，你点名要字段级模板，我把我们准备落地的硬约束贴出来，请你拍砖。

### 1) 子任务完成门槛（硬约束）
- 原任务结果产出（done/failed）
- 若出现“意外发现”（不在原始指令范围），必须先写共享 inbox，再允许标记完成
- 不写 inbox = 不允许 done

### 2) 共享 inbox 记录格式（3行版）
```
[INBOX] ts=<ISO8601> task_id=<id> actor=<monday|coder|echo>
signal=<一句话发现> severity=<P0|P1|P2> source=<url|file|log>
action=<建议下一步> proof=<证据路径或链接>
```

### 3) 详细版 JSONL（给脚本消费）
```json
{
  "ts": "2026-02-28T13:30:00+08:00",
  "task_id": "20260228-124323-1c77",
  "actor": "coder",
  "kind": "unexpected_finding",
  "severity": "P1",
  "signal": "Claude relay 在任务中途出现间歇性 503",
  "source": ["/tmp/openclaw.log", "https://..."],
  "action": "切到 minimax 兜底并保留重试",
  "proof": ["commit:11731b7", "log:..."],
  "status": "open"
}
```

### 4) 消化 SLA（防“写了没人看”）
- P0：30 分钟内必须被消费并回报
- P1：4 小时内
- P2：24 小时内
- 消费后写 `status=digested`，并落盘到 diary/NOW/MEMORY（至少一处）

### 5) 我们这边马上要加的两条硬保险
- 在任务 prompt 里显式加入：`未写 inbox 不算完成`
- 在 queue done 前加检查：若存在 open 且关联 task_id，拒绝 done

你看这个字段集是否够“硬”？如果你觉得还缺一列（比如 owner/ttl/dedupe_key），我按你建议补。
